staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go

```diff
@@ -73,3 +73,44 @@ type CSIDriverSpec struct {
 	// +optional
 	PodInfoRequiredOnMount *bool `json:"podInfoRequiredOnMount"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CSINodeInfo holds information about all CSI drivers installed on a node.
+type CSINodeInfo struct {
+	metav1.TypeMeta `json:",inline"`
+	// ObjectMeta.Name must be node name.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// List of CSI drivers running on the node and their properties.
+	CSIDrivers []CSIDriverInfo `json:"csiDrivers"`
+}
+
+// CSIDriverInfo contains information about one CSI driver installed on a node.
+type CSIDriverInfo struct {
+	// Driver is the name of the CSI driver that this object refers to.
+	// This MUST be the same name returned by the CSI GetPluginName() call for
+	// that driver.
+	Driver string `json:"driver"`
+
+	// ID of the node from the driver point of view.
+	NodeID string `json:"nodeID"`
+
+	// Topology keys reported by the driver on the node.
+	TopologyKeys []string `json:"topologyKeys"`
```

Comment:
Not yet, since code freeze is coming up fast I plan to bring back the CSI proposal after that.

For context, the previous discussion about having a cluster-level call is here: https://github.com/container-storage-interface/spec/pull/257

Reply:
This API is alpha. If we are able to get CSI proposal approved for cluster, we can remove this before beta.

Reply:
Will clarify.